### DISCUSSION TITLE
Schema inferred from function's return signature.

### DISCUSSION THREAD

**Discussion by cgarciae** [fastapi/fastapi#7820] [Questions]
Hey @tiangolo! I've been wondering why FastAPI doesn't use the function's return `->` signature instead of the `response_model` argument? I think it would be a bit more natural to express the return schema like this:

```python
@router.post("/")
def create_worker(
    params: CreateWorker, auth_token: str = Depends(shared.get_authorization)
) -> CreateWorkerResponse:
    # code
    return CreateWorkerResponse(...)
```

**Comment by phy25** [comment]
Duplicate of #620.

**Comment by tiangolo** [comment]
Thanks @cgarciae for reporting back and closing the issue :+1:
